lib/itk/queue/consumer.ex

```diff
@@ -102,15 +102,28 @@ defmodule ITKQueue.Consumer do
 
     case connection() do
       {:ok, connection} ->
-        channel =
-          connection
-          |> Channel.open()
-          |> Channel.bind(queue_name, routing_key)
-
-        Process.monitor(channel.pid)
-
-        {:ok, _} = AMQP.Basic.consume(channel, queue_name, self())
-        %__MODULE__{channel: channel, subscription: subscription}
+        try do
+          channel =
+            connection
+            |> Channel.open()
+            |> Channel.bind(queue_name, routing_key)
+
+          Process.monitor(channel.pid)
+
+          {:ok, _} = AMQP.Basic.consume(channel, queue_name, self())
+          %__MODULE__{channel: channel, subscription: subscription}
+        rescue
+          e ->
+            # Subscribe could timeout
```

Comment:
You could also move the same logic from 129->136 into a separate function, up to you anyway.